### DISCUSSION TITLE
Adding a Progress Bar In the Daily Challenge Suggestions Section.

### DISCUSSION THREAD

**Discussion by suhaib-lone** [Amna-Hassan04/Serenity-Guide#66] [Ideas]
Hi there! I have already raised an issue #64 about this. What i am proposing is a timer after user selects a challenge in Daily Challenge Suggestions section of the website in Calm Space. I can add a progress bar using streamlit's inbuild progress() method which will be in consistent with the project. For example, if the meditation is for ten minutes, it will show a progress bar showing how much time has passed and how much is left using a simple bar.
With this we can remove the "Set a reminder to complete this challenge today." message as the user would be able to see the progress in real time.

![slitprog](https://github.com/user-attachments/assets/35614dc7-e229-411d-b90a-cc2d0db96329)

I think with this new small feature the overall experience of users will be great and beneficial.
Thank You,
Suhaib.

**Comment by Amna-Hassan04** [comment]
I like the idea, I'll assign you the issue you can start work on this, and please make sure the bar looks consistent with the projects own visuals and while making a PR make sure to link the issue so that when i merge the pr it closes the issue by itself.

**Reply by suhaib-lone** [reply]
@Amna-Hassan04 Thank you! I will make sure about what you suggested.
